### ISSUE TITLE
Back button memory usage benchmark 40x regression

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#8429]
Somewhere between https://github.com/flutter/engine/commit/ab0953092743319e0dde3a0c8f582d261c232964 and https://github.com/flutter/engine/commit/e5442c96ae955e0ee8e9accdbc69710c36eb441a we regressed the flutter_gallery__back_button_memory diff_total_kb benchmark by 40x, the end_total_kb benchmark by about 1.3x, and some of the other memory benchmarks by smaller amounts.

The candidate checkins are:
 
* @jason-simmons Return an error to Dart when spawning an isolate with an unsupported URI (https://github.com/flutter/engine/pull/3443)
* @chinmaygarde Remove .pyc files from licenses.golden. (https://github.com/flutter/engine/pull/3445)
* @chinmaygarde Roll Skia revision to 3f81f3703a68755c88f5cc4a87728b98f34c4cd4 (https://github.com/flutter/engine/pull/3438)
* @jason-simmons Adapt the physical model layer for Fuchsia (https://github.com/flutter/engine/pull/3441)
* @chinmaygarde Remove rarely used GLConnection stuff in flow. (https://github.com/flutter/engine/pull/3440)
* @poitch New mozart input events. (https://github.com/flutter/engine/pull/3439) 
* @chinmaygarde Allow plugging in a software backend for rendering in the shell. (https://github.com/flutter/engine/pull/3404)
* @vardhan-cr @chinmaygarde Remove callback from ApplicationController. Kill fidl interface. (https://github.com/flutter/engine/pull/3435)
* @danrubel modify RunFrom<X> methods to return error code rather than call exit (https://github.com/flutter/engine/pull/3428)
* @jason-simmons Physical model layer that can draw shadows for a Material widget (https://github.com/flutter/engine/pull/3424)
* @chinmaygarde Remove texture image. (https://github.com/flutter/engine/pull/3434)
* @chinmaygarde Update skia to 1c8e82b72bfeb1a02297688cf082c365b62705f7 (https://github.com/flutter/engine/pull/3433)

**Comment by jason-simmons** [comment]
The increase in GL memory usage is due to the change to using SkImage::makeTextureImage() for images decoded on the IO thread.  The underlying textures aren't being deleted.

**Comment by chinmaygarde** [comment]
I don't understand how MakeTextureImage would affect the lifecycle of the
underlying SkImage. When the SkImage gets collected, the texture should be
cleaned up as well. If not, we would have had a resource leak earlier too.

**Comment by jason-simmons** [comment]
The texture is now being held by a Skia resource cache after the SkImage is deleted.  This was not the case in the old implementation.

**Comment by chinmaygarde** [comment]
@jason-simmons and I just had a chat about this behavior and a proposed fix. To summarize:
* The `SkImage` returned by `SkImage::MakeTextureImage` is described by a GL texture which is also a part of the GL texture cache associated with the `GrContext` on the resource loading thread.
* When we collect all references to the `SkImage`, we expected the GL texture to be collected eagerly. However, it seems that the way the Skia texture cache operates is to only collect GL textures when its GL texture cache eviction routine is triggered.
* The cache eviction seems to only be triggered when the size of the cache items exceeds that set by `GrContext::setResourceCacheLimits`.
* This meant that even though we had no `SkImage`s left to be collected, their GL textures were still in this cache. This was *not* considered a leak since the textures would have been collected the next time candidates for cache eviction were chosen.
* However, this eviction check never happened because the size of items in the cache never approached its limits. Also, unlike the main rendering `GrContext`, the IO thread context survived suspends and resumes of the application.
* The proposed solution is the set the resource loading thread's GrContext limits to be zero. This would essentially make the collected of the GL texture associated with `SkImage` eager when the image is collected.
* The cache was not helping us earlier because once we got rid of the last `SkImage` reference, we never were able to re-associate the GL texture with another `SkImage` anyway.
* We always ensure we collected all `SkImages` on a thread that has a context available.
